kadirboylu/boylu-router
# Chore - CommitLint, Husky, Eslint and Prettier Setup

## Chore - CommitLint, Husky, Eslint and Prettier Setup

This PR is a chore to setup CommitLint, Husky, Eslint and Prettier. This will help for better code quality and commit message format.

## Related Issue

- None

## Motivation and Context

- CommitLint - To enforce commit message format
- Husky - To create git hooks to check commit message format and it's quality
- Eslint - To enforce code quality
- Prettier - To format code

## How Has This Been Tested?

- Run `pnpm lint` or `pnpm run lint:fix` to check code quality
- Run `pnpm router format` to format code
- Run `git commit -m "commit message"` to check commit message format and it's quality. If it's not good, it will show error message and won't allow to commit.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:





## Checklist:







## Screenshots (if appropriate):
